### ISSUE TITLE
Mock doesn't work after setUpPyfakefs

### ISSUE THREAD

**Issue by CausticYarn** [jmcgeheeiv/pyfakefs#334]
After I run a `fake_filesystem_unittest.TestCase` with the `self.setUpPyfakefs()` method, `unittest.mock` doesn't seem to work correctly in subsequent test cases.  

Here is an example that reproduces the issue:

```python
import os
import unittest.mock

from pyfakefs import fake_filesystem_unittest


class TestFakeFsUnittest(fake_filesystem_unittest.TestCase):
    def setUp(self):
        print('{} Before setUpPyfakefs: {}'.format(self.__class__.__name__,
                                                   os.path))
        self.setUpPyfakefs()
        print('{} After setUpPyfakefs: {}'.format(self.__class__.__name__,
                                                  os.path))

    def test_none(self):
        pass


class TestMockOsPath(unittest.TestCase):
    def test_rename(self):
        print('{} before mock: {}'.format(self.__class__.__name__,
                                          os.path))
        with unittest.mock.patch('os.path'):
            print('{} after mock: {}'.format(self.__class__.__name__,
                                             os.path))
```

When the TestFakeFsUnittest runs first, I get:

```
TestFakeFsUnittest Before setUpPyfakefs: <module 'ntpath' from 'C:\\pyenv\\dev36\\lib\\ntpath.py'>
TestFakeFsUnittest After setUpPyfakefs: <pyfakefs.fake_filesystem.FakePathModule object at 0x000001F8E59645C0>
TestMockOsPath before mock: <module 'ntpath' from 'C:\\pyenv\\dev36\\lib\\ntpath.py'>
TestMockOsPath after mock: <module 'ntpath' from 'C:\\pyenv\\dev36\\lib\\ntpath.py'>
```

But if TestMockOsPath runs first (e.g. by renaming TestFakeFsUnittest to be alphabetically after) I get:

```
TestMockOsPath before mock: <module 'ntpath' from 'C:\\pyenv\\dev36\\lib\\ntpath.py'>
TestMockOsPath after mock: <MagicMock name='path' id='2650908079776'>
TestZFakeFsUnittest Before setUpPyfakefs: <module 'ntpath' from 'C:\\pyenv\\dev36\\lib\\ntpath.py'>
TestZFakeFsUnittest After setUpPyfakefs: <pyfakefs.fake_filesystem.FakePathModule object at 0x0000026936764E10>
```

**Comment by mrbean-bremen** [comment]
That's interesting - I could reproduce it. I didn't understand it yet (seems to be some module caching issue), and the obvious workaround is to put such tests into separate files, but I will have another look later.

**Comment by CausticYarn** [comment]
Having the test in separate files doesn't help (at least when using a test suite to run all tests). I created the above example as the simplest reproduction, but I originally discovered the issue from changing one of my tests to use the setUpPyfakefs call, and having tests in another module fail.

**Comment by mrbean-bremen** [comment]
Hm, ok, you are right - this won't help if this a module caching issue. One (ugly) way to avoid this is to import os inside the test -  but there has to be a better way. I will probably have another look tomorrow.

**Comment by mrbean-bremen** [comment]
Fixed by @absolutelyNoWarranty in #336.

**Comment by mrbean-bremen** [comment]
Now fixed the root cause (missing cleanup) in #340.
